File: src/main/java/raisetech/student/management/data/Student.java
package raisetech.student.management.data;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

@Schema(description = "受講生情報")
@Getter
@Setter

public class Student {

  private Integer id;

  @Size (min = 2, max = 50, message = "名前は2文字以上50文字以下で入力してください。")
  private String name;

  private String kanaName;
  private String nickname;

  @NotBlank(message = "メールアドレスが必要です。")
  @Email(message = "有効なメールアドレスを入力してください。")
  private String emailAddress;

  private String residence;

  @Min(value = 18, message = "年齢は18歳以上である必要があります。")
  @Max(value = 100, message = "年齢は100歳以下である必要があります。")
  private int age;

  private String gender;
  private String remark;
  private boolean wasDeleted;
  }

File: student-management-frontend/src/types/student.ts


export interface Student {
    id?: string;
    name: string;
    kanaName: string;
    nickname: string;
    emailAddress: string;
    residence: string;
    age: number;
    gender: string;
    remark: string;
    wasDeleted: boolean;
}

export interface Course {
    id?: string;
    studentId?: string;
    courseName: string;
    courseStartAt: string;
    courseEndAt: string;
}

export interface ApplicationStatus {
    id?: string;
    courseId?: string;
    applicationStatus: '仮申込' | '本申込' | '受講中' | '受講終了';
}

export interface CourseDetail {
    course: Course;
    applicationStatus: ApplicationStatus;
}

export interface StudentDetail {
    student: Student;
    courseDetails: CourseDetail[];
}

export interface SearchFilters {
    name?: string;
    emailAddress?: string;
    gender?: string;
    courseName?: string;
    applicationStatus?: string;
}

export type Gender = '男性' | '女性' | 'その他';
export type Status = '仮申込' | '本申込' | '受講中' | '受講終了';

export const GENDERS: Gender[] = ['男性', '女性', 'その他'];
export const STATUSES: Status[] = ['仮申込', '本申込', '受講中', '受講終了'];